src/orion/algo/axoptimizer.py

```diff
@@ -0,0 +1,344 @@
+        """
+        super().set_state(copy.deepcopy(state_dict))
+        self._client_state = copy.deepcopy(state_dict.get("_client_state"))
+        self._trials_map = copy.deepcopy(state_dict.get("_trials_map"))
+
+    @contextlib.contextmanager
+    def get_client(self):
+        """Instantiate a new `AxClient` from previous snapshot"""
+        if self._client_state is not None:
+            # Copy client state because `from_json_snapshot` modifies it...
+            client = AxClient.from_json_snapshot(copy.deepcopy(self._client_state))
+        else:
+            client = AxClient(
+                random_seed=self.seed,
+                enforce_sequential_optimization=False,
+                verbose_logging=False,
+            )
+
+            client.create_experiment(
+                parameters=orion_space_to_axoptimizer_space(self.space),
+                choose_generation_strategy_kwargs={
+                    "num_initialization_trials": self.n_initial_trials,
+                    "max_parallelism_override": self.max_trials,
+                },
+                objectives={
+                    "objective": ObjectiveProperties(minimize=True),
+                    **{
+                        o: ObjectiveProperties(minimize=True)
+                        for o in self.extra_objectives
+                    },
+                },
+                outcome_constraints=self.constraints,
+            )
+
+        yield client
+
+        self._client_state = client.to_json_snapshot()
+
+    def suggest(self, num):
+        """Suggest a `num`ber of new sets of parameters.
+
+
+        Parameters
+        ----------
+        num: int, optional
+            Number of trials to suggest. The algorithm may return less than the number of trials
+            requested.
+
+        Returns
+        -------
+        list of trials or None
+            A list of trials representing values suggested by the algorithm. The algorithm may opt
+            out if it cannot make a good suggestion at the moment (it may be waiting for other
+            trials to complete), in which case it will return None.
+
+
+        Notes
+        -----
+        New parameters must be compliant with the problem's domain `orion.algo.space.Space`.
+
+        """
+        trials = []
+        with self.get_client() as client:
+            _trials, _ = client.get_next_trials(num)
+            for trial_index, parameters in _trials.items():
+                parameters = AxOptimizer.reverse_params(parameters, self.space)
+
+                # Ax does not support Fidelity dimension type so fake it with
+                # its max
+                if self.fidelity_index is not None:
+                    # Convert 0-dim arrays into python numbers so their type can
+                    # be validated by Ax
+                    parameters[self.fidelity_index] = float(
+                        self.space[self.fidelity_index].high
+                    )
```

Comment:
Maybe this should be part of `reverse_params`?

Reply:
Is there any value in having `reverse_params` as a class method or can we add in `self` to access `self.fidelity_index` in i?

Reply:
Turning it into a standard method would be fine indeed.